### DISCUSSION TITLE
Phantom "Merge branch 'master'" in Repo Title

### DISCUSSION THREAD

**Discussion by lmstearn** [community/community#36899] [Repositories] [answered]
Hi,
This is a bit awkward, a slight oops with a single tree repo for [PrgLnch](https://github.com/lmstearn/PrgLnch) with one contributor. The repo has always been updated through the web interface.
Today, uploaded a commit with a typo in the commit message through the web interface as usual. Consequently, the original intention was to amend this commit message, so attempted to do it using the VS2019. The full Git was not part of the VS installed packages,, which explained he limited options. Cloned it to desktop, and amended the message in a new commit. But then, Git wanted a pull and push, before this all had to be done in a new commit, and now, somehow, there is created a "merge branch master" with nothing in it.
![Repo](https://user-images.githubusercontent.com/7629854/197215875-f08290cb-5ab3-45d3-9b13-ebf7a7d84556.JPG)
From the graphic, it's probably best to remove the last 3 commits leaving c24A14 at the bottom as the head. The data can then be easily pushed after.
Note that in VS the git.ignore and another git related file were tagged **ignore extension** as they were not requ'd at the Github repo- hoping that didn't upset anything. The Git interface is sure to have improved in VS2022.
In the interim, resetting the repo should be done cleanly through Github Desktop with local and remote uris. Are we good to go with that?
Thanks.

**Comment by ossanna16** [comment]
Hey there @lmstearn and thank you for reporting this! I'm glad you found our community 🙂

**Reply by lmstearn** [reply]
Thanks! It's actually quite nice in here, so won't be coming in to grumble. (too much) :speak_no_evil:

**Comment by spenserblack** [comment, marked as answer]
Because you amended the commit, you created a new history that was different from what was remote. You can kind of think of the remote repository and your local repository as separate branches (`origin/master` and `master` actually are separate branches!). Even if you're just changing the commit message, or even if you change nothing at all, the amended commit is a *new* commit with a new hash, but the same parent commit as the pre-amend commit.

Since the remote and local branch had different histories they conflicted, so you couldn't just push up the amended commit. That's why you needed to merge. And, as you've noticed, because the actual diff of the two commits are the same, so the merge commit doesn't show any changes.

I'd recommend avoiding amending commits *unless you're willing to force-push to the remote.* And you should avoid force-pushing to `master`/`main` as much as possible, unless you're 100% confident in what will happen when you force-push.

**Reply by lmstearn** [reply]
Thanks for the answer, and yes, absolutely in this case it's best to leave well enough alone. :slightly_smiling_face:
 Cloned the repo with Github Desktop and performed a `git reset --hard HEAD~3`. GhD wouldn't allow anything to happen until the fetch on the remote occurred. Wasn't going to fight it by attempting to remove the fetch or pull, or anything else in [examples](https://git-scm.com/docs/git-reset#_examples) before any `force-push`.  Not equipped with the knowhow of dealing with a `force-push` fail.
Also tried `Squash` in GHD, no go, as there was a popup about a merge commit in the selected  items, even when there wasn't. Possibly fixed by a rebase, who knows, as it's back to uploading files directly to the web page.
IMOP, this is the little tragedy of Git, in order to do something trivial like correcting a typo, one has to, well, re-invent the universe.  :woozy_face:
